### DISCUSSION TITLE
How should I couple two external programs based on MOOSE?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#27713] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I want to couple two external programs based on MOOSE, but I don't know exactly how to do it. Should it be done through dynamic linking? How can I control the parameters that I want to couple? Are there any specific examples I can refer to?

Ethan

**Comment by GiudGiud** [comment, marked as answer]
Hello

One of the best examples of coupling external codes is the cardinal code. You can find in on GitHub on the neams-th-coe repository

Guillaume

**Reply by Ethan-xj** [reply]
OK. Thank you Giud
